crates/rolldown/src/ast_scanner/mod.rs

```diff
@@ -228,11 +228,13 @@ impl<'me> AstScanner<'me> {
     self.scopes.get_root_binding(name)
   }
 
+  /// `is_dummy` means if it the import record is created during ast transformation.
   fn add_import_record(
     &mut self,
     module_request: &str,
     kind: ImportKind,
     module_request_start: u32,
+    is_dummy: bool,
```

Comment:
Could you store the `Span` directly? It's more clear than pass both `start` and `is_dummy `. Don't worry about if the size get bigger.

Reply:
There is a concern about that if we keep reusing this error for missing `@import` in css files. If so, storing `span_start` only might not be enough.

Reply:
Adding `span_end` only 4 byptes, but because 8 bytes alignment, the struct will increase 8 bytes. I don't think this is trivial.

Reply:
For `url("xxx")`, I kind of hope that we could emphasize  any range we want instead of only the quoted part.

Reply:
since all diagnostics emitter is context aware (it will also display **n** lines above,  **n** lines below), it is enough for most user. I think it is not worth increasing 8 bytes to only provide a little friendly diagnostics.